apps/grades-sync/src/job.rs

```diff
@@ -0,0 +1,203 @@
+use crate::faculty_repository::FacultyRepository;
+use crate::hkdir::{get_departments, get_grades, get_subjects, map_season_index_to};
+use async_trait::async_trait;
+use itertools::{Itertools};
+use std::cmp::min;
+
+use crate::department_repository::DepartmentRepository;
+use crate::grade_repository::GradeRepository;
+use crate::json::{HkdirDepartment, HkdirGrade, HkdirSubject};
+use crate::subject_repository::SubjectRepository;
+use log::info;
+use regex::Regex;
+
+#[async_trait]
+pub trait JobService: Sync {
+    async fn perform_faculty_synchronization(&self) -> anyhow::Result<()>;
+    async fn perform_subject_synchronization(&self) -> anyhow::Result<()>;
+    async fn perform_grade_synchronization(&self) -> anyhow::Result<()>;
+
+    async fn synchronize_single_faculty(&self, faculty: HkdirDepartment) -> anyhow::Result<()>;
+    async fn synchronize_single_subject(&self, subject: HkdirSubject) -> anyhow::Result<()>;
+    async fn synchronize_single_grade(&self, grade: HkdirGrade) -> anyhow::Result<()>;
+}
+
+pub struct JobServiceImpl<'a> {
+    faculty_repository: &'a dyn FacultyRepository,
+    department_repository: &'a dyn DepartmentRepository,
+    subject_repository: &'a dyn SubjectRepository,
+    grade_repository: &'a dyn GradeRepository,
+}
+
+impl<'a> JobServiceImpl<'a> {
+    pub fn new(
+        faculty_repository: &'a dyn FacultyRepository,
+        department_repository: &'a dyn DepartmentRepository,
+        subject_repository: &'a dyn SubjectRepository,
+        grade_repository: &'a dyn GradeRepository,
+    ) -> Self {
+        Self {
+            faculty_repository,
+            department_repository,
+            subject_repository,
+            grade_repository,
+        }
+    }
+}
+
+const MAX_TASK_COUNT: usize = 100;
+
+#[async_trait]
+impl<'a> JobService for JobServiceImpl<'a> {
+    async fn perform_faculty_synchronization(&self) -> anyhow::Result<()> {
+        info!("performing faculty synchronization");
+        let departments = get_departments().await?;
+        let department_count = departments.len();
+        let _chunks_count = min(department_count / MAX_TASK_COUNT, 1000);
```

Comment:
unused variable? if you want to split it into smaller chunks i would just use [std::slice::Chunks](https://doc.rust-lang.org/std/slice/struct.Chunks.html)